addons/fcs/functions/fnc_keyUp.sqf

```diff
@@ -31,9 +31,9 @@ if (_distance == 0) then {
     ] call EFUNC(common,getTargetDistance); // maximum distance: 5000m, 5m precision
 };
 
-private ["_weaponDirection", "_angleTarget"];
-
-_weaponDirection = _vehicle weaponDirection (_vehicle currentWeaponTurret _turret); // @todo doesn't work for sub turrets
+private ["_weapon", "_weaponDirection", "_angleTarget"];
+_weapon = _vehicle currentWeaponTurret _turret;
```

Comment:
The fcs computes the fire solution for all magazines, not only the current weapon so if you switch weapons solution is already computed. We need to find which weapon is used for each magazine.